app/drivers/kscan/kscan_gpio_direct.c

```diff
@@ -124,6 +125,32 @@ static void kscan_direct_irq_callback_handler(const struct device *port, struct
 }
 #endif
 
+static gpio_flags_t kscan_gpio_get_flags(const struct gpio_dt_spec *gpio, bool active) {
+    gpio_flags_t flags = BIT(0) & gpio->dt_flags;
+    if (!active) {
+        flags |= flags ? GPIO_PULL_UP : GPIO_PULL_DOWN;
+    }
+    return flags;
+}
```

Comment:
Perhaps this should just be getting the *extra* flags, e.g.:

```suggestion
static gpio_flags_t kscan_gpio_get_extra_flags(const struct gpio_dt_spec *gpio, bool active) {
    if (active) {
      return 0;
    }
    return (BIT(0) & gpio->dt_flags) ? GPIO_PULL_UP : GPIO_PULL_DOWN;
}
```

This will work fine, once all the calls are using the DT version of the GPIO configure function.